nomos-mix/message/src/packet.rs

```diff
@@ -135,6 +147,41 @@ impl Packet {
         let new_shared_secret = blinding_factor.diffie_hellman(cur_ephemeral_pubkey);
         x25519_dalek::PublicKey::from(new_shared_secret.to_bytes())
     }
+
+    pub fn to_bytes(&self) -> Vec<u8> {
+        concat_bytes(&[
+            &self.header.ephemeral_public_key.to_bytes(),
+            &self.header.encrypted_routing_info.to_bytes(),
+            &self.payload,
+        ])
```

Comment:
Yeah, I think it would be better to [chain](https://docs.rs/itertools/latest/itertools/macro.chain.html) the iterator and collect the bytes. No need to create the extra array and references.

Reply:
Thank you. [a015485](https://github.com/logos-co/nomos-node/pull/897/commits/a0154853b8f53504fb7fd26e03ea5ab1b471585b)